### PR TITLE
Feature2

### PR DESCRIPTION
**Pull request by Vlad-Andreychik**

### CODE REVIEW

File: clients/handler.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            self.send_msg(
+                "If you wanna sign in choose 1, if you wanna sign up choose 2, print !q for quit")
+            msg = self.receive_message().decode('utf-8')
+            # Sign in
+            if msg == "1":
+                connected = self.authentication(db)
+            # Sign up
+            elif msg == "2":
+                connected = self.registration(db)
+            elif msg == '!q':
```
Comment (akopylov89) [inline clients/handler.py:153]: MessageCodes.QUIT.value

File: clients/handler.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        while connected is False:
+            self.send_service_message(MessageCodes.CODE_LISTEN.value)
+            self.send_msg(
+                "If you wanna sign in choose 1, if you wanna sign up choose 2, print !q for quit")
+            msg = self.receive_message().decode('utf-8')
+            # Sign in
+            if msg == "1":
+                connected = self.authentication(db)
+            # Sign up
+            elif msg == "2":
```
Comment (akopylov89) [inline clients/handler.py:151]: MessageCodes.CODE_SUCCESS.value

File: clients/handler.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        db = db_client.DB('../DB/userdata.db')
+        logging.info(f"[NEW CONNECTION {time.strftime('%H:%M:%S', time.localtime())}] {self.address} try to connect.")
+        connected = False
+        while connected is False:
+            self.send_service_message(MessageCodes.CODE_LISTEN.value)
+            self.send_msg(
+                "If you wanna sign in choose 1, if you wanna sign up choose 2, print !q for quit")
+            msg = self.receive_message().decode('utf-8')
+            # Sign in
+            if msg == "1":
```
Comment (akopylov89) [inline clients/handler.py:148]: MessageCodes.CODE_LISTEN.value

File: clients/handler.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            elif msg == '!q':
+                connected = False
+                logging.info(f"[{self.address}] Disconnected")
+            else:
+                logging.info('Try again')
+
+        # Connected
+        while connected:
+            code = self.receive_message()
+            if code.decode('utf-8') == '!q':
```
Comment (akopylov89) [inline clients/handler.py:162]: MessageCodes.QUIT.value

File: server/server_client.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        """
+        Method that closes server
+        """
+        self.server.close()
+
+
+if __name__ == '__main__':
+    server = Server('', 9009)
+    server.start_listen()
+    while True:
```
Comment (akopylov89) [inline server/server_client.py:54]: while True:
        conn, address = server.accept_connection()
        thread = threading.Thread(target=handler.UserHandler, args=(conn, address))
        thread.start()
        logging.info(
            f"[ACTIVE CONNECTIONS {time.strftime('%H:%M:%S', time.localtime())}] {threading.activeCount() - 1}")

put this logic to server.start_listen()

File: user/user_client.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    def close_connection(self):
+        """
+        Method that closes connection with a server
+        """
+        self.client.close()
+
+
+if __name__ == '__main__':
+    user = User('localhost', 9009)
+    user.connect_to_server()
```
Comment (akopylov89) [inline user/user_client.py:71]: connected = True
    while connected:
        code_from_server = user.receive_message()
        if code_from_server.decode('utf-8') == '!q':
            connected = False
            logging.info(f"Disconnected")
        elif code_from_server.decode('utf-8') == '1':
            message_from_server = user.receive_message()
            logging.info(f"{message_from_server.decode('utf-8')}")
            user.send_message_to_server()
        elif code_from_server.decode('utf-8') == '2':
            authorized = True
            message_from_server = user.receive_message()
            logging.info(message_from_server.decode('utf-8'))
            while authorized:
                choice = input("""
                        Choose an option:
                        1 - Send message to server
                        !q - Disconnect
                        """)
                if choice == '1':
                    user.send_code_message_to_server('1')
                    user.send_message_to_server()
                elif choice == '!q':
                    user.send_code_message_to_server('!q')
                    logging.info('You disconnected from the server')
                    user.close_connection()
                    sys.exit()
                else:
                    logging.info("Try again")



put this logic to  user.connect_to_server()

File: clients/handler.py (hunk 1 above)
Comment (Vlad-Andreychik) [reply clients/handler.py:153]: Done

File: clients/handler.py (hunk 2 above)
Comment (Vlad-Andreychik) [reply clients/handler.py:151]: Done

File: clients/handler.py (hunk 3 above)
Comment (Vlad-Andreychik) [reply clients/handler.py:148]: Done

File: clients/handler.py (hunk 4 above)
Comment (Vlad-Andreychik) [reply clients/handler.py:162]: Done

File: server/server_client.py (hunk 5 above)
Comment (Vlad-Andreychik) [reply server/server_client.py:54]: Done

File: user/user_client.py (hunk 6 above)
Comment (Vlad-Andreychik) [reply user/user_client.py:71]: Done

**Pull request closed without merging**